### PR TITLE
rgw: fix collect()'s return in coroutine

### PR DESCRIPTION
**Pull request by cfanz**
Return true when the rgw_spawned_stack is not fully checked.
Signed-off-by: Xinying Song <songxinying@cloudin.cn>

### CODE REVIEW

File: src/rgw/rgw_coroutine.cc [hunk 1]
```diff
@@ -324,15 +324,15 @@ bool RGWCoroutinesStack::collect(RGWCoroutine *op, int *ret, RGWCoroutinesStack
       *ret = r;
       ldout(cct, 20) << "collect(): s=" << (void *)this << " stack=" << (void *)stack << " encountered error (r=" << r << "), skipping next stacks" << dendl;
       new_list.insert(new_list.end(), ++iter, s->entries.end());
-      done &= (iter != s->entries.end());
+      need_retry |= (iter != s->entries.end());
```
Comment (tianshan) [inline src/rgw/rgw_coroutine.cc]: '='  is enough and more clear, since next break will jump out of the loop

Comment (cfanz) [reply src/rgw/rgw_coroutine.cc]: Yes, you're right 👍. I have modified it.

**Comment by cfanz** [comment]
@tianshan Thanks for your advice, I have corrected it.

**Comment by yuriw** [comment]
test this please

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-02-14-2124](https://trello.com/c/guwy7LoB/261-wip-yuri3-testing-2018-02-14-2124)

**Pull request merged**